src/main/java/org/assertj/core/internal/Failures.java

```diff
@@ -119,7 +120,7 @@ public AssertionError failure(AssertionInfo info, AssertionErrorFactory factory)
   public AssertionError failure(AssertionInfo info, ErrorMessageFactory messageFactory) {
     AssertionError error = failureIfErrorMessageIsOverridden(info);
     if (error != null) return error;
-    AssertionError assertionError = new AssertionError(messageFactory.create(info.description(), info.representation()));
+    AssertionError assertionError = new AssertionFailedError(messageFactory.create(info.description(), info.representation()));
```

Comment:
we can't do that because it won't work for JUnit 4 users, instead `AssertionErrorCreator` should be used as it handles creating `AssertionFailedError` without explicit dependency on JUnit 5 (it works if it's there and fallback to JUnit 4 or simple `AssertionError` otherwise).
`Failures.failure(AssertionInfo info, ErrorMessageFactory messageFactory, Object actual, Object expected)` can also be used as it calls `AssertionFailedError`.